lib/facter/custom_facts/util/normalization.rb

```diff
@@ -16,8 +19,10 @@ class NormalizationError < StandardError; end
       # @return [void]
       def normalize(value)
         case value
-        when Integer, Float, TrueClass, FalseClass, NilClass, Symbol, Date
+        when Integer, Float, TrueClass, FalseClass, NilClass, Symbol
           value
+        when Date, Time
+          value.iso8601
```

Comment:
In facter 3, `Time.now` seems to more closely resemble `#to_s` rather than `#iso8601`. For `Date` it seems to be the same... so maybe just call `to_s` on them?

```
2.6.3 :001 > Facter.add(:custom_time) { setcode { Time.now } }
 => #<Facter::Util::Fact:0x000055d8ee5de608> 
2.6.3 :002 > Facter.value(:custom_time)
 => 2021-03-18 15:25:16 +0200 
```

But since we didn't fully "support" these type of facts before, I'm ok with converting them to ISO 8601 if the user doesn't do the normalization.

Reply:
Seems like `.to_s` calls `iso8601` on Date https://apidock.com/ruby/Date/to_s

Reply:
hm... I would call .to_s on both then, just to be consistent

Reply:
I would keep iso8601 as is the standard for working with datetimes. Calling `to_s` we rely on Ruby to delegate to the right methods, which can change/be different depending on the ruby version. Also having a standardised format would make it easier for someone that wants to parse the resulted time string(back to time objects), in another programming language.

Reply:
Makes sense to me 👍